src/bindings/python/flux/job/list.py

```diff
@@ -96,12 +103,22 @@ def get_job(self):
         return self.get()["job"]
 
     def get_jobinfo(self):
+        """
+        Returns a JobInfo object for the job.
+        """
         return JobInfo(self.get_job())
 
 
 # list-id is not like list or list-inactive, it doesn't return an
 # array, so don't use JobListRPC
 def job_list_id(flux_handle, jobid, attrs=["all"]):
+    """Watch a job by providing an ID. 
+
+    This is a non-blocking function that allows users to asynchronously
+    get job event data. Returns a RPC future that is fulfilled each time 
+    the job enters a new state.
+    :rtype: JobListIdRPC
```

Comment:
There may be some confusion here. This function (and the corresponding `job-list.list-id` RPC) send a single response with the information the job-list module has for the job at that moment. It is not multi-response and does not send updates on each state change. (Perhaps you are confusing this with the `event_watch_async` interface, which sends a response for each event (not each state though))

Perhaps just remove this block and instead say simply:
"Sends an RPC to the job-list module to query information about the provided jobid. Use the `get_job()` or `get_jobinfo()` method on the returned `JobListIdRPC` to obtain the job data as a dict or a `JobInfo` object respectively" 

(an RPC is always asynchronous so mentioning that the function is non-blocking is probably not warranted here)

Reply:
You're right that I confused what this was doing with`event_watch`, both here and in the `job_list_id` function docstring...fixed. Thanks.